### ISSUE TITLE
Partial mocking on superclass is not consistent depending on method signature

### ISSUE THREAD

**Issue by phylexx** [jmockit/jmockit1#384]
* Version of JMockit that was used: 
1.24 & 1.30

* Description of the problem:
Given the following three classes :
** Parent
** ChilA extends Parent
** ChildB extends Parent
When partial mocking an instance of ChildA,
Then method calls to Parent class using an instance of ChildB does not always proceed to real implementation, it depends on the method signature.

See the attached test case (using junit 4.x and jmockit 1.30).
[InconsistentPartialMockingTest.zip](https://github.com/jmockit/jmockit1/files/710520/InconsistentPartialMockingTest.zip)

When _ExpectedInvocation.getDefaultValueForReturnType_ return a not null object, no cascadingValue is returned in _RecordAndReplayExecution.defaultReturnValue_ and the real implementation is called after exiting _recordOrReplay_.
When _ExpectedInvocation.getDefaultValueForReturnType_ return a null object,  and _RecordAndReplayExecution.defaultReturnValue_ will get a cascadedValue and real implementation will not be called while exiting _recordOrReplay_.

In other words, when JMockit knows how to instantiate a defaultReturnValue for the called method, the real implementation is not called, otherwise it returns Void.class and real implementation is called.

I would expect that the real implementation is called on a non mocked instance/class when calling a method in its superclass whatever its prototype.
